### ISSUE TITLE
Async functions cause tf_upgrade_v2 to crash

### ISSUE THREAD

**Issue by TheBicPen** [tensorflow/tensorflow#35103]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Manjaro Linux (kernel 4.19)
- TensorFlow installed from (source or binary): binary? I'm not sure, I used `pip install tensorflow`
- TensorFlow version: 2.0.0
- Python version: Python 3.6.8
- Installed using virtualenv? pip? conda?: pip. This is in a virtualenv environment though.

**Describe the problem**
Running `tf_upgrade_v2` on a file that has async functions causes it to crash. 

**Provide the exact sequence of commands / steps that you executed before running into the problem**
```tf_upgrade_v2 --infile ./image_classification/test_tf.py --outfile ./image_classification_v2/test_tf.py
Traceback (most recent call last):
  File "/home/alex/git-repos/bic-bot-py/bin/tf_upgrade_v2", line 8, in <module>
    sys.exit(main())
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/tensorflow_core/tools/compatibility/tf_upgrade_v2_main.py", line 139, in main
    args.input_file, output_file, upgrade)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/tensorflow_core/tools/compatibility/tf_upgrade_v2_main.py", line 40, in process_file
    upgrader.process_file(in_filename, out_filename)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/tensorflow_core/tools/compatibility/ast_edits.py", line 900, in process_file
    temp_file)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/tensorflow_core/tools/compatibility/ast_edits.py", line 960, in process_opened_file
    self.update_string_pasta("".join(lines), in_filename))
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/tensorflow_core/tools/compatibility/ast_edits.py", line 916, in update_string_pasta
    t = pasta.parse(text)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/__init__.py", line 25, in parse
    annotator.visit(t)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 1201, in visit
    super(AstAnnotator, self).visit(node)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 133, in visit
    super(BaseVisitor, self).visit(node)
  File "/home/alex/.pyenv/versions/3.6.8/lib/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 47, in wrapped
    f(self, node, *args, **kwargs)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 225, in visit_Module
    self.generic_visit(node)
  File "/home/alex/.pyenv/versions/3.6.8/lib/python3.6/ast.py", line 261, in generic_visit
    self.visit(item)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 1201, in visit
    super(AstAnnotator, self).visit(node)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 133, in visit
    super(BaseVisitor, self).visit(node)
  File "/home/alex/.pyenv/versions/3.6.8/lib/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/home/alex/.pyenv/versions/3.6.8/lib/python3.6/ast.py", line 261, in generic_visit
    self.visit(item)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 1201, in visit
    super(AstAnnotator, self).visit(node)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 133, in visit
    super(BaseVisitor, self).visit(node)
  File "/home/alex/.pyenv/versions/3.6.8/lib/python3.6/ast.py", line 253, in visit
    return visitor(node)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 47, in wrapped
    f(self, node, *args, **kwargs)
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 673, in visit_Return
    self.token('return')
  File "/home/alex/git-repos/bic-bot-py/lib/python3.6/site-packages/pasta/base/annotate.py", line 1340, in token
    token_val, token.src, token.start[0], token.line))
pasta.base.annotate.AnnotationError: Expected 'return' but found 'async'
line 1: async def f():
```

**Any other info / logs**
test_tf.py:
```
async def f():
    return
```

**Comment by gadagashwini** [comment]
@TheBicPen, Could you post the complete standalone code to reproduce the issue. Thanks!

**Comment by TheBicPen** [comment]
Here is the command I ran: `tf_upgrade_v2 --infile test_tf.py --outfile test_tf_v2.py `. The contents of the file are the same as I posted above:
```
async def f():
    return
```
It also crashed when I tested with a file containing more code (same command as above) :
```
from image_classification import image_classify as ic

tf_sess = None
classifications = None


def start_tf():
    global tf_sess
    global classifications
    classifications = ic.get_classifications()
    if tf_sess is not None:
        print("TensorFlow session already exists")
        return "TensorFlow session already exists"
    tf_sess = ic.main()
    return "started TensorFlow session"


def stop_tf():
    global tf_sess
    global classifications
    tf_sess.close()  # free resources used by the tf session
    tf_sess = None  # remove the reference to it
    classifications = None
    print("ended TensorFlow session")
    return "ended TensorFlow session"


async def classify_attachment(message, tf_sess, classifications):
    if classifications is None:
        return "image classifications not found"
    if tf_sess is None:
        return "TensorFlow session not found"
    if len(message.attachments) == 0:
        return "no valid attachments in message"

    # check image using image-classification
    attachment = message.attachments[0]
    print("message contains attachment at:{0}".format(attachment.url))
    img = await attachment.read()
    result = ic.classify_image(img, tf_sess, classifications)
    return (result, attachment)
```

**Comment by gadagashwini** [comment]
@TheBicPen, Looks like `image_classification` is not defined and tensorflow import is missing. Please provide the more information to reproduce the issue. Thanks!

**Comment by TheBicPen** [comment]
I don't have access to my computer right now, but the specific code that caused the issue is here: https://github.com/TheBicPen/bic-bot-py/tree/master/image_classification . The command I ran was `tf_upgrade_v2 --intree image_classification --outtree image_classification_v2`. I ran this command from the root of the repo. Additionally, the `test_tf.py` file that I used to try to isolate the issue is located here: https://github.com/TheBicPen/bic-bot-py/blob/master/test/test_tf.py . I hope this information is enough to reproduce the issue

**Comment by gadagashwini** [comment]
@TheBicPen, Tried reproducing the reported issue with provided code.But I didn't get error message.
Please find the [gist](https://colab.sandbox.google.com/gist/gadagashwini/3e5e9ba8c8982b276584c3176919334d/untitled325.ipynb) and confirm the issue. Thanks!

**Comment by TheBicPen** [comment]
@gadagashwini, The issue can be reproduced as seen [here](https://colab.research.google.com/gist/TheBicPen/4f32482213a0797f0e4a9d4c2228a913/untitled325.ipynb).

**Comment by gowthamkpr** [comment]
This is an issue in google_pasta, a library we use to write that script in. This is fixed as shown in the following [comment](https://github.com/tensorflow/tensorflow/issues/26486#issuecomment-497027128). Please go though it. Thanks!

**Comment by TheBicPen** [comment]
Same issue occurs with pasta 0.1.8:
```
>pip show google-pasta
Name: google-pasta
Version: 0.1.8
Summary: pasta is an AST-based Python refactoring library
Home-page: UNKNOWN
Author: Nick Smith
Author-email: smithnick@google.com
License: Apache 2.0
Location: ....\appdata\local\programs\python\python36\lib\site-packages
Requires: six
Required-by: tensorflow
```
This is on Windows this time.

**Comment by gowthamkpr** [comment]
@TheBicPen Lets make things clear here. If you use pasta 0.1.8 are you still facing the issue in Manjaro Linux?

**Comment by TheBicPen** [comment]
I am facing the same issue with pasta 0.1.8 on Windows. I haven't checked yet on Manjaro.

**Comment by gowthamkpr** [comment]
Initially the issue has been created for **Manjaro Linux** So, lets stick to that for now. If you are looking into windows, please create a new issue for windows. Thanks!

**Comment by TheBicPen** [comment]
> Initially the issue has been created for **Manjaro Linux** So, lets stick to that for now. If you are looking into windows, please create a new issue for windows. Thanks!

I had a hardware change over the holidays so I only have a Windows machine now. I'll spin up a Manjaro VM tonight, but I'm pretty sure that the same issue occurs across both platforms.

**Comment by TheBicPen** [comment]
```
...
pasta.base.annotate.AnnotationError: Expected 'message' but found 'async'
line 29: async def classify_attachment(message, tf_sess, classifications):

[manjaro@manjaro-i3 bic-bot-py]$ pip show google-pasta
Name: google-pasta
Version: 0.1.8
Summary: pasta is an AST-based Python refactoring library
Home-page: UNKNOWN
Author: Nick Smith
Author-email: smithnick@google.com
License: Apache 2.0
Location: /home/manjaro/.local/lib/python3.7/site-packages
Requires: six
Required-by: tensorflow
```
Same issue on Manjaro.

**Comment by gowthamkpr** [comment]
Its failing with pasta 0.1.7 too..!!

**Comment by tomerk** [comment]
Pasta has been upgraded to 0.2 and this should be fixed now.
